prometheus_alerting_consistency.md

```diff
@@ -0,0 +1,103 @@
+# Prometheus Alerting Consistency
+
+## Summary
+
+This guide is originally based on the [Alerting Consistency][1] proposal for OpenShift alerts.
+
+Clear and actionable alerts are a key component of a smooth operational experience.
+Ensuring you have clear and concise guidelines for engineers and SRE creating new alerts will result in a better experience for end users.
+
+## Recommended Reading
+
+A list of references on good alerting practices:
+
+* [Google SRE Book - Monitoring Distributed Systems][2]
+* [Prometheus Alerting Documentation][3]
+* [Alerting for Distributed Systems][4]
+
+## Style Guide
+
+* Alert names MUST be CamelCase, e.g.: `BackendServiceStuck`
+* Alert names SHOULD be prefixed with a component, e.g.: `ZookeeperPersistentVolumeFillingUp`
+  * There may be exceptions for some broadly scoped alerts, e.g.: `TargetDown`
+* Alerts MUST include a `severity` label indicating the alert's urgency.
+  * Valid severities are: `critical` or `warning` — see below for
+    guidelines on writing alerts of each severity.
+* Alerts MUST include `summary` and `description` annotations.
+  * Think of `summary` as the first line of a commit message, or an email
+    subject line.  It should be brief but informative.  The `description` is the
+    longer, more detailed explanation of the alert.
+* Alerts SHOULD include a `namespace` label indicating the source of the alert.
+  * Many alerts will include this by virtue of the fact that their PromQL
+    expressions result in a namespace label.  Others may require a static
+    namespace label.
+* All alerts MUST include a `sop_url` annotation.
```

Comment:
... which directs the SRE to the location of the SOP (or runbook) that is specific to fixing that problem.

Reply:
```suggestion
* All alerts MUST include a `sop_url` annotation which directs the SRE to the location of the SOP (or runbook) that is specific to fixing that problem.
```

Reply:
@david-martin I might say "which directs the responder(s) to the location..." What do you think?

Reply:
Why `sop_url`, instead of using the upstream `runbook_url` (e.g. as discussed [here][1])?  If an alert has both an `sop_url` and a `runbook_url`, how does an SRE know which to check first, and if/when to check the other?

[1]: https://github.com/prometheus-operator/kube-prometheus/issues/731

Reply:
@wking I am impartial here. I've mainly seen `sop_url`, but have also seen `runbook_url`.
Is the defacto out in the wild `runbook_url`?
We can easily change it, and add an explanation saying it's the existance of a field that links to what to do that is more important than the name of the field.

Reply:
Alternate wording suggestion:
```suggestion
* All alerts MUST include an annotation (e.g. `runbook_url`) which directs the responder(s) to the location of the runbook (or SOP) that is specific to fixing that problem.
```